go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -464,7 +504,24 @@ func createNodeManagementPorts(name string, nodeAnnotator kube.Annotator, waiter
 		}
 		config.OvnKubeNode.MgmtPortRepresentor = rep
 	}
-	ports := NewManagementPorts(name, subnets)
+
+	if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+		link, err := getManagmentPortNetDev(config.OvnKubeNode.MgmtPortNetdev)
```

Comment:
This looks unoptimal.
Call to this function at `Create()` will always succeed on first `LinkByName()` unless no internal error happened. 
For DPUHost mode - because it is invoked 1st time here.
For Full mode - because device existence verified in if block just above this one.

How I suggest to rewrite.
Drop this call completely and get deviceID in the same way it is done for the Full mode. In this way we assure that device exists and we know it's netdevice name. Need to change `exportManagementPortAnnotation()` to get deviceID for `config.OvnKubeNode.MgmtPortNetdev` first and if failed, get for `types.K8sMgmtIntfName` and overwrite `MgmtPortNetdev` accordingly. In `Create()` replace `getManagmentPortNetDev()` (BTW, it is "manag**e**ment") with a single call `LinkByName(config.OvnKubeNode.MgmtPortNetdev)`.

Reply:
It's unclear to me what you mean by getting the deviceID in the saw as for Full mode?

We don't know the netdev name. In some cases it's not equal to the provided name in vf management port config. Nor is it equal to the mp0 name. That's why we need to get the netdev and use link instead.

Reply:
yes you can. The current code does exactly that. It gets the name from netlink.Link. It even renames the interface if the name attribute is incorrect.  Please see the `Create` function in magagement-port-dpu.go, more specifically the `setName := link.Attrs().Name != types.K8sMgmtIntfName` variable stores if we have to change the name or not.

Reply:
This is what I'm talking about. `setName := link.Attrs().Name != types.K8sMgmtIntfName` is just to make sure that we don't rename netdevice when it is already renamed to `ovn-k8s-mp0`. We could remove this line and always rename mgmt port even if it already have `ovn-k8s-mp0` name.

Reply:
It's possible that link.Attrs().Name != the provided config and also != mp0. The check is not: 

```
if link.Attrs().Name != config.OvnKubeNode.MgmtPortNetdev && link.Attrs().Name != types.K8sMgmtIntfName
  ...
```

On my system `link.Attrs().Name` is something else (`net...`) and it gets renamed here. Hence, we need to keep handle the case that it's something else. That's why I've extracted that piece of code and now use it to detect the link name too.

Reply:
Here are the logs:

```
I0417 10:56:35.744152  899798 default_node_network_controller.go:124] Enable node proxy healthz server on 0.0.0.0:10256
I0417 10:56:35.744166  899798 default_node_network_controller.go:541] Starting the default node network controller
I0417 10:56:35.750830  899798 default_node_network_controller.go:597] Node worker-52 ready for ovn initialization with subnet 10.128.2.0/23
I0417 10:56:35.751050  899798 default_node_network_controller.go:448] Exporting management port annotation
I0417 10:56:35.751261  899798 management-port-dpu.go:188] Lookup netdevice link and existing management port
I0417 10:56:35.751326  899798 management-port-dpu.go:215] Setup netdevice management port: net041723105631
I0417 10:56:35.968159  899798 ovs.go:200] Exec(1): /usr/sbin/sysctl -w net.ipv4.conf.ovn-k8s-mp0.forwarding=1
I0417 10:56:35.968685  899798 ovs.go:203] Exec(1): stdout: "net.ipv4.conf.ovn-k8s-mp0.forwarding = 1\n"
I0417 10:56:35.968703  899798 ovs.go:204] Exec(1): stderr: ""
I0417 10:56:35.975248  899798 gateway_init.go:407] Initializing Shared Gateway Functionality on DPU host
I0417 10:56:35.984949  899798 helper_linux.go:69] Provided gateway interface "ens1f0np0", found as index: 6
I0417 10:56:35.987759  899798 helper_linux.go:94] Found default gateway interface ens1f0np0 192.168.122.1
```

As you can see, `management-port-dpu.go:215` shows that at that time, the management port name is `net041723105631` and the code then proceeds to rename it since `setName := link.Attrs().Name != types.K8sMgmtIntfName` will be `true`. `netLinkOps.LinkByName` returns this. That's why my patch needs to use the same code.

Reply:
Does `config.OvnKubeNode.MgmtPortNetdev == "net041723105631"`?

Reply:
no, it's `ens..`

Reply:
K8sMgmtIntfName  and MgmtPortNetdev are 2 different things 
K8sMgmtIntfName   is constant [1]  which Create will rename to and MgmtPortNetdev  is the  name of the device before the rename to K8sMgmtIntfName. So what dmytro is suggesting is to call the getManagmentPortNetDev as you did, but after calling NewManagementPorts you will have the struct of the management ports [2] and you can put the netdvice name that you already find above. 
The code in [3] can be removed because we already know that is the management port net device name. 

[1] -  https://github.com/ovn-org/ovn-kubernetes/blob/cb70545d59c83a1a9e3756a229d6ffc8db1fa293/go-controller/pkg/types/const.go#L12
[2] -https://github.com/ovn-org/ovn-kubernetes/blob/cb70545d59c83a1a9e3756a229d6ffc8db1fa293/go-controller/pkg/node/management-port-dpu.go#L169-L172
[3] - https://github.com/ovn-org/ovn-kubernetes/blob/cb70545d59c83a1a9e3756a229d6ffc8db1fa293/go-controller/pkg/node/management-port-dpu.go#L186-L195

Reply:
On my system, there are **3** names in play:

1. ovn-k8s-mp0 (which is what we want as the eventual name)
2. ens1f0v0 (which is what the user thinks the name will be of the netdev in the container at startup and which should be renamed to ovn-k8s-mp0)
3. net041723105631 (the actual name that you get when you add debug code to show the contents of `/sys/class/net/` just before [`setName` ](https://github.com/ovn-org/ovn-kubernetes/blob/cb70545d59c83a1a9e3756a229d6ffc8db1fa293/go-controller/pkg/node/management-port-dpu.go#L213))

Number 3 above is the reason why we are having this discussion. Without taking care of that case, the proposed solution doesn't work. I don't know why the name has changed to that (it clearly includes the date and time). I've just written the code in a way that uses the same logic that causes the rename to succeed in https://github.com/ovn-org/ovn-kubernetes/blob/cb70545d59c83a1a9e3756a229d6ffc8db1fa293/go-controller/pkg/node/management-port-dpu.go#L182.

Reply:
@bn222, are You switching mgmt port netdevices? Case 3 name comes because You had used, lets say, 1st VF as mgmt port, then switched to 2nd VF. Because of that, old one was renamed to this generic name (ovnkube-kube just doesn't have information about original name of the old mgmt port). And then You switched mgmt port back to the 1st VF, which have this awkward name.

Reply:
I'm not switching mgmt port.

The device is created with sriov network operator and then it is passed to ovn-k.

Reply:
Look here
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/management-port_linux.go#L444
That is why I'm saying mgmt port device had been changed.

Reply:
Aha, I understood why it happens. Your changes broke the logic, so ovnkube always do a renaming of the "old" mgmt port, even though there is no old management port. I will put a comment there.